File: services/lucy-dotnet/src/Lucy.AuthPayment.Api/Modules/Payment/Models/Gift.cs
namespace lucy_auth_payment.Modules.Payment.Models;

public class Gift
{
    public string GiftId { get; set; } = null!;

    public string GiftName { get; set; } = string.Empty;

    public decimal PriceAmount { get; set; }

    public string? IconUrl { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.Now;
}

File: lucy-web-stack/lucy-backend/src/main/java/com/lucy/backend/payment/entity/Gift.java
package com.lucy.backend.payment.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "Gifts")
@Data
public class Gift {
    @Id
    @Column(name = "GiftId", length = 50)
    private String giftId;

    @Column(name = "GiftName", length = 100, nullable = false)
    private String giftName;

    @Column(name = "PriceAmount", precision = 12, scale = 2, nullable = false)
    private BigDecimal priceAmount;

    @Column(name = "IconUrl", length = 255)
    private String iconUrl;

    @Column(name = "IsActive", nullable = false)
    private Integer isActive = 1;

    @Column(name = "CreatedAt", updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}
